iceoryx_posh/include/iceoryx_posh/internal/popo/base_client.inl

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2020 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_POPO_BASE_CLIENT_INL
+#define IOX_POSH_POPO_BASE_CLIENT_INL
+
+#include "iceoryx_posh/internal/popo/base_client.hpp"
+
+namespace iox
+{
+namespace popo
+{
+// ============================== BaseClient ============================== //
+
+template <typename Port>
+inline BaseClient<Port>::BaseClient(const capro::ServiceDescription& service,
+                                    const ClientOptions& clientOptions) noexcept
+    : m_port(*iox::runtime::PoshRuntime::getInstance().getMiddlewareClient(service, clientOptions))
```

Comment:
`getMiddlewareClient` can return a `nullptr` but before it does this it calls the errorHandler with a severe error type. Therefore, we do not have to handle the `nullptr` here - right?

I am just loud thinking since this bugs me a bit - but you do not have to handle `nullptr` here.

Reply:
Exactly. We should change this for all ports to return an expected instead of just terminating. Then the user can terminate.